packages/headless/src/controllers/result-list/headless-result-selection-helpers.ts

```diff
@@ -0,0 +1,62 @@
+import {Result} from '../../api/search/search/result';
+import {Engine} from '../../app/headless-engine';
+import {ResultAnalyticsActions} from '../../features/analytics';
+
+// 1 second is a reasonable amount of time to catch most longpress actions
+const defaultDelay = 1000;
+
+/**
+ * Helpers that more easily allow sending analytics when a search result is selected.
+ */
+export type ResultSelectionHelpers = ReturnType<
+  typeof buildResultSelectionHelpers
```

Comment:
I would like us to improve the name. I really like @olamothe's suggestion `buildResultLink`.
My second choice is `buildResult` which I find more agnostic (does not mention `Link`), but admittedly is also more abstract, and maybe harder to understand.

Reply:
I already started renaming it to `buildResultLinkAnalytics`, but if you don't like it I can change it.

Reply:
Tbh, I prefer one of the other names over `buildResultLinkAnalytics`